### PR TITLE
Unpair from device when removing.

### LINKED ISSUE #350: Unpairing Devices
As a user, I want to unpair devices associated with the gateway

### PR DESCRIPTION
**Pull request by mrstegeman**
The actual unpairing process is left up to the adapter.

Fixes #350

### CODE REVIEW

**Comment by hobinjk** [comment]
Testing with virtual-things-adapter the first DELETE request didn't complete and the second errored with "Failed to remove thing virtual-things-0: Remove already in progress". This may be a side effect of vta using an out-of-date gateway-addon-node or something.

**Comment by mrstegeman** [comment]
@hobinjk Can you try with this: https://github.com/mozilla-iot/gateway-addon-node/pull/1

**Comment by mrstegeman** [comment]
@hobinjk I did confirm that the gateway-addon PR fixed this issue.

**Comment by mrstegeman** [comment]
@dhylands Not currently. Z-Wave will continue to behave as it always has for now.

**Comment by benfrancis** [comment]
This is arguably better than what we have now but I'm still not sure if it's the right UX. But I don't yet have a better suggestion.

File: src/test/integration/things-test.js [hunk 1]
```diff
@@ -396,20 +396,6 @@ describe('things/', function() {
       }
     }
     expect(!found).assert('should not find thing in /things output');
-
-    res = await chai.request(server)
```
Comment (hobinjk) [inline src/test/integration/things-test.js]: Why isn't the thing expected here? It looks like it should still be there but I might be wrong about how mock-adapter works

Comment (mrstegeman) [reply src/test/integration/things-test.js]: There was previously some weirdness in that things could be present in the adapters but not in the gateway. Due to the changes in this patch set, that is no longer the case.

**Pull request merged**